_source/_posts/2023-08-30-passwordless-authentication.md

```diff
@@ -0,0 +1,203 @@
+
+FIDO's Client to Authenticator Protocol complements WebAuthn to create a seamless, easy-to-use, secure authentication experience. 
+
+{% img blog/passwordless-authentication/webauthn-ctap.jpg alt:"Icons representing WebAuthn in the browser and CTAP devices with a heart in between" width:"800" %}{: .center-image }
+
+A new term for this authentication mechanism without passwords is often called "passwordless." And the combination of WebAuthn and CTAP2 hardware devices to enable passwordless authentication is called a "passkey." What a great name!
+
+## FIDO2 authentication with WebAuthn and CTAP
+ 
+So how does this work? At a high level, your FIDO2 authentication flow utilizes each of the three factors:
+1. Something you know, a PIN to unlock the device
+2. Something you have, the authenticator device, which may be a security key or a capable phone
+3. Something you are, biometric data such as fingerprint or facial recognition
+
+Before we authenticate, we need an account with compliant credentials, so let's start with the registration process. We'll use a public demo site demonstrating how passkeys work as we step through what's going on in the registration and authentication process. Follow along at [webauthn.io](https://webauthn.io/).
+
+There are a couple of components at play. From the user's perspective, they interact with a website and a device like their phone. From a technology perspective, there's WebAuthn and CTAP2. I'll show screenshots of the user perspective underlying network calls and describe the hand-off between WebAuthn and CTAP2.
+
+## Registering a user for passkeys
+
+The user opens their favorite site in the browser. In the FIDO2 world, the site that makes the WebAuthn calls is the _Relying Party_, and the browser is the _Client_. If your site uses a single-page application frontend, the relying party includes both the frontend and backend server. 
+
+{% img blog/passwordless-authentication/webauthnio.jpg alt:"webauthn.io site showing a user name field an buttons to register or authenticate" width:"800" %}{: .center-image }
+
+The web app for the site displays a form to gather user information, an email address, or a user name. The user enters the required information and presses **Register**. The web app recognizes the browser as FIDO2 capable and asks if the user wants to sign in using a passkey instead of a password. Of course, the user says, "Yes!"
+
+{% img blog/passwordless-authentication/webauthnio-register.jpg alt:"webauthn.io site showing a user name entered in the form and a modal from the browser asking to create a passkey" width:"800" %}{: .center-image }
+
+The web app calls the server with credential request options, such as the username and registration-related data. You can open debugging capabilities in your browser and watch the network requests. Here, we see the payload for the call, which includes the username.
```

Comment:
can open debugging capabilities be a link to how to open the developer console somewhere? if we catch someone in the lucky 10k of devtools they may have a rough time figuring out what "open debugging capabilities" means

Reply:
It depends on the browser, so hence, no explicit instructions.